### DISCUSSION TITLE
How to install searx using a subdomain

### DISCUSSION THREAD

**Discussion by winux1** [searx/searx#2480] [Installation issue: Official wiki] [answered]
Hello! 
I would like to know how to install searx using a subdomain for example searx.domain.tld not domain.tld/searx using apache and the install scripts

Thanks for your help in advance!

**Comment by return42** [comment]
untested: 

- set `PUBLIC_URL="https://searx.example.org/"` in [.config](https://github.com/searx/searx/blob/master/.config.sh)
- install searx, morty and filtron using [installation scripts](https://searx.github.io/searx/admin/installation.html#installation-scripts)

Compare [architecture](https://searx.github.io/searx/admin/architecture.html)

It is filtron and morty you need to add a reverse proxy configuration for in your apache setup.

The Apache setup really depends on your distribution, we talk about different distributions at https://searx.github.io/searx/admin/installation-apache.html#apache-reverse-proxy but set up a subdomain in appache you have to define a 

``` apache
<VirtualHost *:443>
	ServerName searx.example.org
...
```

To this virtual host, you have to add a Location sections for morty and filtron

``` apache
<Location / >
    Require all granted
    Allow from all
    ProxyPreserveHost On
    ProxyPass http://127.0.0.1:4004
    RequestHeader set X-Script-Name /
</Location>

<Location /morty >
    Require all granted
    Allow from all
    ProxyPass http://127.0.0.1:3000
    RequestHeader set X-Script-Name /morty
</Location>
```
If you have a doubt about the values of the ProxyPass URLs, you can use  `FILTRON_LISTEN` and `MORTY_LISTEN`

```sh

$ ./utils/filtron.sh --help
...
  PUBLIC_URL     : https://searx.example.org/
  PUBLIC_HOST    : searx.example.org
  SERVICE_USER   : filtron
  FILTRON_TARGET : 127.0.0.1:8888
  FILTRON_API    : 127.0.0.1:4005
  FILTRON_LISTEN : 127.0.0.1:4004

$ ./utils/morty.sh --help
...
  PUBLIC_URL_MORTY:     https://searx.example.org/morty/
  MORTY_LISTEN:         127.0.0.1:3000
  SERVICE_USER:         morty
...
```

.. hope that helps.

**Reply by winux1** [reply]
Hi @return42!
Thanks for your help, but unfortunately, it didn't work, I'm still geting this error: Too many redirects

**Comment by unixfox** [comment]
searx-docker allow hosting Searx easily under a subdomain, but you will have to stop using apache for that: https://github.com/searx/searx-docker

**Reply by winux1** [reply]
Thanks for your advice, but I can't to use docker on my server.
I would like to do something like this: https://search.privacytools.io/

**Reply by unixfox** [reply]
> Thanks for your advice, but I can't to use docker on my server.
> I would like to do something like this: [search.privacytools.io](https://search.privacytools.io/)

Why can't you use Docker on your server?

**Reply by winux1** [reply]
Because I only have 1Gb ram

**Reply by unixfox** [reply]
> Because I only have 1Gb ram

Docker runs fine with just 1 GB of RAM. It runs even fine with just 512 MB of RAM.

**Reply by winux1** [reply]
Is it working with raspberry pi?

**Reply by unixfox** [reply]
> Is it working with raspberry pi?

Unfortunately not yet. There is an issue for that here: https://github.com/searx/searx-docker/issues/27

**Reply by winux1** [reply]
So can you help me how to install searx with apache on subdomain?

**Reply by unixfox** [reply]
> So can you help me how to install searx with apache on subdomain?

I don't use apache, but I found a tutorial for installing Searx with apache on any domain/subdomain: https://dbxit.com/simple-install-for-searx-on-ubuntu-with-apache. It doesn't matter if that's a tutorial for Ubuntu 18.04 by the way.

**Reply by return42** [reply]
> Too many redirects

check your apache files .. are there left any entries or file from a previous attempt?

> Is it working with raspberry pi?

The  [installation scripts](https://searx.github.io/searx/admin/installation.html#installation-scripts) are not tested under pi!!!

You have to install manually ..

> but I found a tutorial for installing Searx with apache on any domain/subdomain

I can't recommend the instructions under the given link and BTW it is not talking about subdomain and does not install morty & filtron.

I recommend to use https://searx.github.io/searx/admin/installation-searx.html to setup searx, morty & filtron.

About setting up a subdomain search for [apache pi subdomain](http://darmarit.org/searx/search?q=apache%20pi%20subdomain&categories=general&language=en-US) .. but most often you will find [ `<VirtualHost *:443>` examples like the one I give above](https://github.com/searx/searx/discussions/2480#discussioncomment-292640).

**Reply by winux1** [reply]
I will stay on domain.tld/searx for now

**Comment by winux1** [comment, marked as answer]
Finally I was able to solve my issue. Now my `searx.conf` looks like this:
```apache
LoadModule proxy_module         /usr/lib/apache2/modules/mod_proxy.so
LoadModule proxy_http_module    /usr/lib/apache2/modules/mod_proxy_http.so

<VirtualHost *:80>
Servername domain.tld
ServerAlias www.domain.tld
ProxyPreserveHost On
ProxyPass /morty/ http://127.0.0.1:3000/
ProxyPassReverse /morty/ http://127.0.0.1:3000/
ProxyPass / http://127.0.0.1:4004/
CustomLog /dev/null combined
</VirtualHost>
```
Thanks for everyone who tried to help!
